osgp/protocol-adapter-oslp/web-device-simulator/src/main/java/org/opensmartgridplatform/webdevicesimulator/web/controller/DeviceManagementController.java

```diff
@@ -378,39 +378,34 @@ public String showEditDevice(@PathVariable final Long deviceId, final Model mode
 
   @PostMapping(value = DEVICE_EDIT_URL)
   public String editDevice(
-      @SuppressWarnings("squid:S4684") @ModelAttribute(MODEL_ATTRIBUTE_DEVICE)
-          final Device
-              updated, // webdevicesimulator doesn't have to be as strict in this Sonar issue
+      // web device simulator doesn't have to be as strict in this Sonar issue
+      @SuppressWarnings("squid:S4684") @ModelAttribute(MODEL_ATTRIBUTE_DEVICE) final Device updated,
       @PathVariable final Long deviceId,
       final BindingResult bindingResult,
       final RedirectAttributes attributes,
       final Model model) {
 
-    if (bindingResult.hasErrors()) {
-      return DEVICE_EDIT_VIEW;
+    if (!bindingResult.hasErrors()) {
```

Comment:
An early return is encouraged over nesting if statements.

Reply:
Hmm, maybe this could be refactored in another way?